api/src/task/task.py

```diff
@@ -63,6 +63,10 @@ def increment(self, name: str, value: int = 1, prefix: str | None = None) -> Non
 
         self.metrics[name] += value
 
+        if prefix is not None:
+            # Rather than re-implement the above, just re-use the function without a prefix
+            self.increment(f"{prefix}.{name}", value, prefix=None)
```

Comment:
Making sure I understand this code correctly. When you add a metric with a prefix, you are actually creating two metrics. One with the prefix, and one without, right?

Reply:
Yes, that way we can have high-level and granular metrics all in one.

So if we were working with a metric like `records_processed`, and were working with record types `a`, `b` and `c` we could end up with something like:
```
records_processed -> 10
a_records_processed -> 5
b_records_processed -> 3
c_records_processed -> 2
```
Without needing several separate calls to the function.